cms/djangoapps/contentstore/views/course.py

```diff
@@ -899,8 +899,7 @@ def settings_handler(request, course_key_string):
                     ee_min_score_pct = request.json.get('entrance_exam_minimum_score_pct', None)
 
                     # If the entrance exam box on the settings screen has been checked,
-                    # and the course does not already have an entrance exam attached...
-                    if entrance_exam_enabled and not course_entrance_exam_present:
+                    if entrance_exam_enabled:
                         # Load the default minimum score threshold from settings, then try to override it
```

Comment:
Right -- I think we need to decouple the creation of the exam and the
setting/changing of the score.
On Feb 17, 2015 2:23 AM, "Asad" <notifications@github.com> wrote:

> In cms/djangoapps/contentstore/views/course.py
> <https://github.com/edx/edx-platform/pull/6997#discussion_r24797292>:
>
> > @@ -899,8 +899,7 @@ def settings_handler(request, course_key_string):
> >                      ee_min_score_pct = request.json.get('entrance_exam_minimum_score_pct', None)
> >
> >                      # If the entrance exam box on the settings screen has been checked,
> > -                    # and the course does not already have an entrance exam attached...
> > -                    if entrance_exam_enabled and not course_entrance_exam_present:
> > +                    if entrance_exam_enabled:
> >                          # Load the default minimum score threshold from settings, then try to override it
>
> Hmmm, matt i think i should revert back this condition into its original
> state like as >
>
> if entrance_exam_enabled and not course_entrance_exam_present:
> Because during creating an entrance exam , we are first cleaning out any
> old entrance exams in entrance_exam.py line 84.
> So this condition > if entrance_exam_enabled: -- will create a new
> instance of an entrance exam instead of updating the minimum passing score
> of a pre created exam.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/6997/files#r24797292>.
>